drivers/pinctrl/pfc_rcar.c

```diff
@@ -21,19 +21,23 @@
 #if defined(CONFIG_SOC_SERIES_RCAR_GEN3)
 #define PFC_RCAR_GPSR 0x100
 #define PFC_RCAR_IPSR 0x200
-DEVICE_MMIO_TOPLEVEL_STATIC(pfc, DT_DRV_INST(0));
-static uintptr_t reg_base[1];
 #elif defined(CONFIG_SOC_SERIES_RCAR_GEN4)
 #define PFC_RCAR_GPSR 0x040
 #define PFC_RCAR_IPSR 0x060
+#else
+#error Unsupported SoC Series
+#endif
+
 /* swap both arguments */
 #define PFC_REG_ADDRESS(idx, node_id) DT_REG_ADDR_BY_IDX(node_id, idx)
-static const uintptr_t reg_base[] = {
+static uintptr_t reg_base[] = {
 	LISTIFY(DT_NUM_REGS(DT_DRV_INST(0)), PFC_REG_ADDRESS, (,), DT_DRV_INST(0))
 };
-#else
-#error Unsupported SoC Series
-#endif
+
+#define PFC_REG_SIZE(idx, node_id) DT_REG_SIZE_BY_IDX(node_id, idx)
```

Comment:
please use here `DT_INST_REG_SIZE_BY_IDX`

Reply:
Should I also use the `DT_INST_*` macro for the `PFC_REG_ADDRESS`?

Reply:
`DT_NUM_REGS` doesn't seem to have it's `INST` version, so that one is not necessary even though it would be appreciated if you add it since it's a trivial thing

Reply:
changed for both macros, --- reg size and addr